src/components/Details/Tabs/Evolution.js

```diff
@@ -1,149 +1,50 @@
-		'use-item': 'Use',
-
-	};
-
-	// load evolution on mount
-	useEffect( () => {
-		setLoading( true );
-
-		fetchPokemonEvolutionChain( pokemon.id ).then( ( data ) => {
-			setEvolutionChain( [] );
-			setCurrentEvolution( data.chain );
-			setLoading( false );
-		} );
-
-		// eslint-disable-next-line
-    }, [ pokemon ]);
-
-	useEffect( () => {
-		getNextEvolution();
-
-		// eslint-disable-next-line
-    }, [ currentEvolution ])
-
-	// get next pokemon evolution
-	const getNextEvolution = () => {
-		if ( currentEvolution.length === 0 || currentEvolution.evolves_to.length === 0 )
-			return null;
-
-		const nextEvolution = currentEvolution.evolves_to[ 0 ];
-		const details = nextEvolution.evolution_details[ 0 ];
-
-		// extract useful data
-		const current = currentEvolution.species.name;
-		const next = nextEvolution.species.name;
-		const trigger = triggersDisplayName[ details.trigger.name ];
-		const triggerValue = details.min_level || details.min_happiness || details.item?.name.replace( '-', ' ' ) || '';
-		const currentId = extractId( currentEvolution.species.url );
-		const nextId = extractId( nextEvolution.species.url );
-
-		// get images URL
-		const currentImage = getImageURL( currentId );
-		const nextImage = getImageURL( nextId );
-
-		// set current evolution to next evolution
-		setCurrentEvolution( ( prev ) => ( prev.evolves_to[ 0 ] ) );
-
-		// push new evolution object
-		setEvolutionChain( ( prev ) => {
-			return [ ...prev, {
-				current,
-				next,
-				trigger,
-				triggerValue,
-				currentId,
-				nextId,
-				currentImage,
-				nextImage,
-			} ];
-		} );
-	};
-
-	// get pokemon image by id
-	const getImageURL = ( id ) => {
-		// base URL for images
-		const baseURL = 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other';
-
-		// has svg
-		if ( parseInt( id ) <= 649 ) {
-			return `${ baseURL }/dream-world/${ id }.svg`;
-		}
-		// has only png
-		return `${ baseURL }/official-artwork/${ id }.png`;
-	};
-
-	// fetch pokemon from evolution chain and show its details
-	const fetchPokemon = ( name ) => {
-		if ( ! name || pokemon.name === name ) {
-			return;
-		}
-
-		// get pokemon data & show it
-		setLoading( true );
-		fetchPokemonData( name ).then( ( data ) => {
-			changePokemon( data );
-		} );
-	};
+	const { isLoading, evolutionChain } = useEvolutionChain( pokemon.id );
 
 	return (
 		<div className="tab tab-evolution">
 
 			<h3>Evolution Chain</h3>
 
-			{ loading &&
+			{ isLoading &&
                 <Loader />
 			}
 
-			{ ! loading && evolutionChain.length === 0 &&
+			{ ! isLoading && evolutionChain.length === 0 &&
                 <div>
-                    This pokemon does'nt evolove
+                    This pokemon doesn't evolove
                 </div>
 			}
 
-			{ ! loading &&
-				evolutionChain.map( ( e, i ) => {
+			{ ! isLoading &&
+				evolutionChain.map( ( { currentId, currentName, currentImage, nextId, nextName, nextImage, trigger, triggerValue }, i ) => {
```

Comment:
Seems like the inner content of this map should be a separated component, that gets each map value as a prop.

Reply:
Done